src/sentry/api/endpoints/project_app_store_connect_credentials.py

```diff
@@ -151,6 +131,7 @@ def post(self, request, project):
             }
             credentials["encrypted"] = encrypt.encrypt_object(encrypted, key)
             credentials["type"] = "appStoreConnect"
+            credentials["refreshDate"] = datetime.utcnow()
             credentials["id"] = uuid4().hex
```

Comment:
Updating the refresh date here is probably pretty safe (although technically incorrect).
We should remember when we got a success message from the 2fa (that's when the session was created).

I would still store the date in the session at the end of the 2fa call and retrieve it from there, saving the date at the end of 2fa is more important in the update call thought (see comment there).